src/main/java/Duke.java

```diff
@@ -1,10 +1,134 @@
+import java.io.IOException;
+
+/**
+ * Duke is the main class of the Duke chatbot.
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage(filePath);
+        try {
+            //Load from storage
+            tasks = new TaskList(storage.loadData());
+        } catch (Exception e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public static void main(String[] args) throws IOException, DukeException {
+        new Duke("data/tasks.txt").run();
+    }
+
+    public void run() throws IOException, DukeException {
+        ui.welcome();
+        String fullCommand = ui.readCommand();
+        String newString = "";
+        String[] splitBySpace;
+        int num;
+        Task t;
+        String desc;
+        String by;
+
+        while (!fullCommand.equals("bye")) {
+            try {
+                Command cmd = parser.parseCommand(fullCommand);
+                newString = "";
+                desc = "";
+                by = "";
+
+                if (cmd != null) {
+                    switch (cmd) {
+                    case LIST:
```

Comment:
I think it might be clearer if the newlines are consistent throughout the switch case block, for eg only having newlines before the next case.